daemon/cmd/daemon_main.go

```diff
@@ -1795,24 +1796,47 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 	return nil
 }
 
-func registerEndpointStateResolver(lc cell.Lifecycle, daemonPromise promise.Promise[*Daemon], endpointRestorer *endpointRestorer, resolver promise.Resolver[endpointstate.Restorer]) {
-	var wg sync.WaitGroup
+func registerEndpointStateResolvers(lc cell.Lifecycle, endpointRestorer *endpointRestorer, endpointsRestoredResolver promise.Resolver[endpointstate.EndpointsRestored], initialPoliciesComputedResolver promise.Resolver[endpointstate.InitialPoliciesComputed]) {
```

Comment:
nit: reads very verbose ~ for types like `promise.Resolver[endpointstate.EndpointsRestored]` I think I'm fine with a parameter name that doesn't duplicate all of that info. so `epsRestored` would suffice IMO. You'll call `Resolve` on it anyway.